atomic_reactor/utils/resource_manager.py

```diff
@@ -6,193 +6,304 @@
 of the BSD license. See the LICENSE file for details.
 """
 
-import os
-import logging
-import json
-import paramiko
 import backoff
+import json
 import jsonschema
-from typing import List, Any, Dict, Tuple
+import logging
+import paramiko
+import random
+import os
+from contextlib import contextmanager
 from datetime import datetime
 from shlex import quote
+from typing import List
+
+RESOURCE_LOCK_SCHEMA = 'schemas/resource-lock-content.json'
+SSH_COMMAND_TIMEOUT = 45
+TIMEOUT_FAIL_BUILD = 3000
+RELATIVE_PATH = 'resource_manager'
+RESOURCE_INFO_JSON = 'info.json'
+RETRY_ON_SSH_EXCEPTIONS = (paramiko.ssh_exception.NoValidConnectionsError,
+                           paramiko.ssh_exception.SSHException)
+BACKOFF_FACTOR = 5
+MAX_RETRIES = 3
 
-PARAMIKO_MAX_RETRIES = 3
-PARAMIKO_BACKOFF_FACTOR = 1
+logger = logging.getLogger('resource-manager')
+logger.setLevel(logging.ERROR)
 
-ROOT_PATH = 'resource_manager/'
 
-RESOURCE_LOCK_SCHEMA = 'schemas/resource-lock-content.json'
+class LockError(Exception):
+    """ Attempt for resource locking returned non-zero exitcode """
+
+
+class LockRetry(Exception):
+    """ Retry locking, no resources were available """
+
```

Comment:
IMO we should inherit from RuntimeError (this is currently used in OSBS everywhere)
```suggestion
class ResourceManagerError(RuntimeError):
   """Base module exception"""

class LockError(ResourceManagerError):
    """ Attempt for resource locking returned non-zero exitcode """


class LockRetry(ResourceManagerError):
    """ Retry locking, no resources were available """

```

Reply:
Added